### PR TITLE
(enhancement)types: add support to specify ImagePullSecrets

### PR DESCRIPTION
**Pull request by ksatchit**
Signed-off-by: ksatchit <karthik.s@mayadata.io>

**What this PR does / why we need it**:

- Support for pulling chaos resource (chaos-runner & experiment pod) images from private registries

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #https://github.com/litmuschaos/litmus/issues/2128

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] Labelled this PR & related issue with `documentation` tag
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests

### CODE REVIEW

**Review by chandankumar4 — DISMISSED** [review]
lgtm

**Comment by rahulchheda** [comment]
@ksatchit Could you add a little context, I'm unable to understand the need for this. An issue would be great.

**Review by chandankumar4 — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Comment by ksatchit** [comment]
> @ksatchit Could you add a little context, I'm unable to understand the need for this. An issue would be great.

Have linked the issue. This PR adds support for pulling litmus images that are being maintained in private registries (which is typical with enterprise usage). In such cases, a Kubernetes secret is created with the registry cred (username/password/email) which is referred to in the podspec attribute `imagePullSecret`.

**Review by ispeakc0de — APPROVED** [review]

**Review by uditgaurav — APPROVED** [review]

File: pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go [hunk 1]
```diff
@@ -267,6 +269,7 @@ type ExperimentComponents struct {
 	Secrets               []Secret                    `json:"secrets,omitempty"`
 	ExperimentAnnotations map[string]string           `json:"experimentannotation,omitempty"`
 	ExperimentImage       string                      `json:"experimentImage,omitempty"`
+	ExperimentImagePullSecrets []corev1.LocalObjectReference `json:"experimentImagePullSecrets,omitempty"`
```
Comment (rahulchheda) [inline pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go:272]: nit: this line, doesn't seem to be linted

**Review by rahulchheda — APPROVED** [review]
/lgtm

**Pull request merged**